projects/pgai/pgai/sqlalchemy/__init__.py

```diff
@@ -0,0 +1,175 @@
+from dataclasses import asdict
+from typing import Any, Generic, TypeVar
+
+from pgvector.sqlalchemy import Vector  # type: ignore
+from sqlalchemy import ForeignKey, Integer, MetaData, Text
+from sqlalchemy.orm import DeclarativeBase, Mapped, backref, mapped_column, relationship
+
+from pgai.configuration import (
+    ChunkingConfig,
+    DiskANNIndexingConfig,
+    EmbeddingConfig,
+    HNSWIndexingConfig,
+    NoScheduling,
+    ProcessingConfig,
+    SchedulingConfig,
+)
+
+# Type variable for the parent model
+T = TypeVar("T", bound=DeclarativeBase)
+
+
+class EmbeddingModel(DeclarativeBase, Generic[T]):
+    """Base type for embedding models with required attributes"""
+
+    embedding_uuid: Mapped[str]
+    id: Mapped[int]
```

Comment:
We aren't constrained to an integer `id` foreign key. How should we handle that? Source tables may have compound primary keys of various types

Reply:
Oh you're right. I'll try make this a generic, but I'm a bit scared of breaking strict typing. I don't want the embedding class to have a possibly infinite amount of Any fields.